src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
Yeah I figured that you wanted to mutate the object later on, but I don't get really why you would want to instantiate any controller with an `undefined` reference. I would instantiate whatever needs this once it's there?